projects/fsastorefrontlib/src/cms-components/user-profile/user-profile.component.ts

```diff
@@ -0,0 +1,107 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  OnDestroy,
+  OnInit,
+} from '@angular/core';
+import { combineLatest, Observable, Subscription } from 'rxjs';
+import { User, UserAccountFacade } from '@spartacus/user/account/root';
+import { RoutingService, UserIdService } from '@spartacus/core';
+import { map } from 'rxjs/operators';
+
+import {
+  FSUserRole,
+  FSUser,
+  InsuranceQuoteList,
+} from '../../occ/occ-models/occ.models';
+import { ConsentService } from '../../core/my-account/facade/consent.service';
+import { QuoteService } from '../../core/my-account/facade';
+import { PolicyService } from '../../core/my-account/facade';
+import { ClaimService } from '../../core/my-account/facade';
+
+@Component({
+  selector: 'cx-fs-user-profile',
+  templateUrl: 'user-profile.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class UserProfileComponent implements OnInit, OnDestroy {
+  constructor(
+    protected userAccountFacade: UserAccountFacade,
+    protected fsConsentService: ConsentService,
+    protected routingService: RoutingService,
+    protected userIdService: UserIdService,
+    protected quoteService: QuoteService,
+    protected policyService: PolicyService,
+    protected claimService: ClaimService
+  ) {}
+
+  private subscription = new Subscription();
+  customer$: Observable<FSUser>;
+  seller: boolean;
+  customerQuotes$: Observable<InsuranceQuoteList>;
+  customerPolicies$: Observable<any>;
+  customerClaims$: Observable<any>;
+  assets: { [key: string]: any }[];
+  assetSelected: string;
+
+  ngOnInit(): void {
+    this.loadCustomerDetails();
+  }
+
+  loadCustomerDetails() {
+    this.subscription.add(
+      combineLatest([
+        this.routingService.getRouterState(),
+        this.userIdService.getUserId(),
+        this.userAccountFacade.get(),
+      ])
+        .pipe(
+          map(([routingData, userId, user]) => {
+            const customerId = routingData?.state?.params?.customerId;
+            if (customerId) {
+              this.getCustomerAssets(user, userId, customerId);
+            } else {
+              this.getSellerAssets();
+            }
+          })
+        )
+        .subscribe()
+    );
+  }
+
+  protected getCustomerAssets(user: User, userId: string, customerId: any) {
```

Comment:
Should naming be different? Here we are fetching policies/claims/quotes for the user that is seller, it's little confusing

Reply:
Renamed